octoprint_displayprogress/__init__.py

```diff
@@ -5,16 +5,36 @@
 import octoprint.events
 
 class DisplayProgressPlugin(octoprint.plugin.ProgressPlugin,
-                            octoprint.plugin.EventHandlerPlugin,
-                            octoprint.plugin.SettingsPlugin):
+							octoprint.plugin.EventHandlerPlugin,
+							octoprint.plugin.SettingsPlugin,
+							octoprint.plugin.TemplatePlugin,
+							octoprint.plugin.AssetPlugin):
 
 	##~~ SettingsPlugin
 
 	def get_settings_defaults(self):
 		return dict(
+			barsymbol="pound",
 			message="{bar} {progress:>3}%"
-		)
+			)
+
+	##~~ TemplatePlugin
+
+	def get_template_configs(self):
+		return [
+			dict(type="settings", custom_bindings=False)
+		]
+
+	def get_template_vars(self):
+		return dict(barsymbol=self._settings.get(["barsymbol"]))
```

Comment:
Using template variables for this settings value is a bad idea. The problem is the following: Template vars are part of the cached HTML of a page. Changing a settings doesn't necessarily invalidate the cache. Meaning that if you inject the setting value as a template variable, it won't necessarily be properly updated in the frontend. E.g. the user sets it to tilde instead, but the frontend doesn't properly reflect that.